src/ClientImpl.cpp

```diff
@@ -281,12 +284,18 @@ ClientImpl::ClientImpl(ClientConfig config) throw(voltdb::Exception, voltdb::Lib
     }
 #endif
     struct event_config *cfg = event_config_new();
+    if (!cfg)
+    {
+        throw voltdb::LibEventException();
+    }
     event_config_set_flag(cfg, EVENT_BASE_FLAG_NO_CACHE_TIME);//, EVENT_BASE_FLAG_NOLOCK);
     m_base = event_base_new_with_config(cfg);
     assert(m_base);
     if (!m_base) {
+	event_config_free(cfg);
         throw voltdb::LibEventException();
     }
+    event_config_free(cfg);
```

Comment:
cfg needs to be freed in destructor instead. Change cfg to be class member variable.